app/src/androidTest/java/com/example/sharingang/MapActivityTest.kt

```diff
@@ -0,0 +1,67 @@
+package com.example.sharingang
+
+import android.os.Build
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.GrantPermissionRule
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiSelector
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+@RunWith(AndroidJUnit4::class)
+class MapActivityTest {
+    @get:Rule
+    val activityRule = ActivityScenarioRule(MapActivity::class.java)
+    //@get:Rule 
+    //var permissionRule: GrantPermissionRule = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
+    // code taken from https://alexzh.com/ui-testing-of-android-runtime-permissions/
+    fun grantPermission() {
```

Comment:
Make this helper private
```suggestion
    private fun grantPermission() {
```